transport/server.cc

```diff
@@ -461,33 +461,53 @@ future<foreign_ptr<std::unique_ptr<cql_server::response>>>
             tracing::set_response_size(trace_state, response->size());
             return response;
         },  utils::result_catch<exceptions::unavailable_exception>([&] (const auto& ex) {
+            clogger.warn("{}: request resulted in unavailable_error, stream {}, code {}, message [{}]",
+                _client_state.get_remote_address(), stream, ex.code(), ex.what());
             try { ++_server._stats.errors[ex.code()]; } catch(...) {}
             return make_unavailable_error(stream, ex.code(), ex.what(), ex.consistency, ex.required, ex.alive, trace_state);
         }), utils::result_catch<exceptions::read_timeout_exception>([&] (const auto& ex) {
+            clogger.warn("{}: request resulted in read_timeout_error, stream {}, code {}, message [{}]",
+                _client_state.get_remote_address(), stream, ex.code(), ex.what());
             try { ++_server._stats.errors[ex.code()]; } catch(...) {}
             return make_read_timeout_error(stream, ex.code(), ex.what(), ex.consistency, ex.received, ex.block_for, ex.data_present, trace_state);
         }), utils::result_catch<exceptions::read_failure_exception>([&] (const auto& ex) {
+            clogger.warn("{}: request resulted in read_failure_error, stream {}, code {}, message [{}]",
```

Comment:
Those messages will definitely spam the logs and they usually happen in batches. This should be at `trace` log level.

Reply:
The rule (which I am sure is not always followed) is that if an event happens as a result of a client request it should be logged at the `trace` level.

Reply:
OK

Reply:
If something is useful "for debugging", I guess its level should be "debug".

I think that as long as we don't have any rare-limiting in Scylla logging (I think we don't, right?) default-on logging (and that would be the case for "warn" level) is dangerous for the types of errors that can be caused by the user - not by a Scylla bug. The best example I gave below are the "overload" message. Imagine that the user sends 100,000 requests per second and Scylla decides that's too fast and wants to respond to 10,000 of those with the "overloaded" error. Do we really want to see 10,000 warning lines in the log at that point? I don't think so.

Also, it's not like before your patch we didn't know that these events were happening - we had **counters** for these events even before your patch. So another question I had for you is what does the log message add - even for debugging. Is knowing the specific client IP address that sent the request helpful? What kind of Scylla bug can those log messages help understand beyond the fact that those events happened and are counted?

Reply:
> If something is useful "for debugging", I guess its level should be "debug".
> 

If something is not useful "for debugging" it should probably not be printed at any level.

Reply:
> Do we really want to see 10,000 warning lines in the log at that point? I don't think so.

I've already agreed to lower the log level to trace or debug.

> So another question I had for you is what does the log message add - even for debugging. Is knowing the specific client IP address that sent the request helpful? What kind of Scylla bug can those log messages help understand beyond the fact that those events happened and are counted?

Counters are mode difficult to use for debugging than the logs:
  * you have to know that they exists and how to query them;
  * difficult to correlate with other events;
  * missing details (client ip, error message, etc).

And I don't see arguments not to log these message with trace or debug. Production installations shouldn't run with these levels.

Reply:
The "you have to know that they exist" argument doesn't convince me - the same would be true for debug-level log messages (I understand that's why you wanted warn-level log messages, but I think we agreed that that's too much).

I agree that lowering the level makes it much harder to argue against these additions. I can still curious when and why these additional details can help debugging in any way (as @gleb-cloudius said, if it doesn't help debugging, it shouldn't be logged, at any level). I guess the issue of "correlating with other events" is a good explanation why it was helpful to you for debugging..

Reply:
>  I can still curious when and why these additional details can help debugging in any way

I debugged [session.execute](https://github.com/scylladb/scylla-dtest/pull/2903/files#diff-23b7f74fc428b588ed84c89057961218ddeef449707ebb1e15c684a958696e18R314) resulting in "broken pipe" on the client. There were no additional clues in neither Scylla nor client logs. I noticed the counters in the code, could they have helped me? Maybe, if I had known the code thoroughly. But I hadn't, and it was much easier for me to add those logs instead of trying to guess what was really happening by he counters. My guess is that the next person who runs into similar problems will also not be thoroughly familiar with the code (it is quite complex, given all the asynchrony), and he will have to go through this path again. These logs could have saved him some time.

Regarding the problem above, the CQL query turned out to be too large, it [ended with the error](https://github.com/scylladb/scylladb/pull/11949/files#diff-5ddc61dd27612d0e2d48deba0325d00a24bf8d14d0b05de0f3b4126f4632236fR642), but this error was not delivered to the client. Since we did not read the query text from the incoming socket, the next request in the same session tried to treat the remains of the previous request as the beginning of the next one and broke down on this. This led to a breakdown in the driver, and it failed to properly report the original problem.